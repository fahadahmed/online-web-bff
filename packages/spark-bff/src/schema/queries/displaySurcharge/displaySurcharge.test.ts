import { createTestClient, gql } from 'test/apolloTestUtils';

test('returns all the displaySurcharge data', async () => {
  const { query } = createTestClient();
  const res = await query({
    query: gql`
      query {
        displaySurcharge(
          input: { operationTypes: [PAY_BILL, SETUP_MONTHLY_PAYMENT] }
        ) {
          surchargeValues {
            operationType
            masterCard
            visa
            amex
            dinersClub
            displayText
          }
        }
      }
    `,
  });

  expect(res.errors).toBeUndefined();
  expect(res.data!.displaySurcharge).toMatchSnapshot();
});
