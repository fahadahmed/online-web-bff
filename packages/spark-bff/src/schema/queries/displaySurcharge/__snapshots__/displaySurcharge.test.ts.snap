// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns all the displaySurcharge data 1`] = `
Object {
  "surchargeValues": Array [
    Object {
      "amex": "2%",
      "dinersClub": "2%",
      "displayText": Array [
        "American Express and Diners Club incur a 2% surcharge.",
        "Visa/MasterCard, Prezzy Card, international credit, debit or charge cards incur a 3% surcharge.",
      ],
      "masterCard": "3%",
      "operationType": "PAY_BILL",
      "visa": "3%",
    },
    Object {
      "amex": "2%",
      "dinersClub": "2%",
      "displayText": Array [
        "Automated payments using a credit/debit card incurs a 1.5% surcharge (plus GST, if any). Surcharge applies.",
      ],
      "masterCard": "3%",
      "operationType": "SETUP_MONTHLY_PAYMENT",
      "visa": "3%",
    },
  ],
}
`;
